ql/src/java/org/apache/hadoop/hive/ql/parse/ReplicationSemanticAnalyzer.java

```diff
@@ -393,35 +394,22 @@ private void dumpFunctionMetadata(String dbName, Path dumpRoot) throws SemanticE
     }
   }
 
-  /**
-   *
-   * @param ast
-   * @param dbName
-   * @param tblName
-   * @param dbRoot
-   * @return tbl dumped path
-   * @throws SemanticException
-   */
-  private Path dumpTbl(ASTNode ast, String dbName, String tblName, Path dbRoot) throws SemanticException {
-    Path tableRoot = new Path(dbRoot, tblName);
+  private void dumpTable(ASTNode ast, String dbName, String tblName, Path dbRoot)
+      throws SemanticException {
     try {
-      URI toURI = EximUtil.getValidatedURI(conf, tableRoot.toUri().toString());
       TableSpec ts = new TableSpec(db, conf, dbName + "." + tblName, null);
-
-      ExportSemanticAnalyzer.prepareExport(ast, toURI, ts, getNewReplicationSpec(), db, conf, ctx,
-          rootTasks, inputs, outputs, LOG);
+      TableExport.Paths exportPaths = new TableExport.Paths(ast, dbRoot, tblName, conf);
```

Comment:
I agree the functionality of Paths is different but it is still closely related to TableExport. It is not generic functionality to be shared by any other classes. So, it is better to keep it internal to TableExport.
About too many variables, probably need to use TableSpec to store some variables like Hive db, table name etc. Also, ast is used only for debug logging and shall recheck if it can be removed.